### ISSUE TITLE
is it possible to have permanent URLs for tarball downloads?

### ISSUE THREAD

**Issue by sam-github** [openssl/openssl#3556]
Node.js just experienced build failures requiring a change to our scripts to adjust to the move of `openssl-fips-2.0.9.tar.gz` to `old/fips/`, see https://github.com/nodejs/node/pull/13172#issuecomment-304044213

Would it be possible to keep the .tar.gz files in a permanent directory structure, or in some other way arrange for them to have a permanent URI? Perhaps they could be organized by the top two numbers in the release stream to avoid so many files in one directory that its impossible to find the latest ones? The Web pages would still just point to the most recent, to make it easy for humans to find the latest with a browser.

Node has done this, see https://nodejs.org/dist/, and the stable URLs have enabled a set of thirdparty tools for downloading, building, installing, etc that automatically notice new releases, and don't require patching as archives are demoted from "current" to "old".

cf. https://github.com/nodejs/node/issues/13219

**Comment by richsalz** [comment]
The simplest thing for us to do, @levitte is to copy the new release to the old directory right away.  I just did it by hand for all the release streams.

**Comment by sam-github** [comment]
People might still link to the new ones and not realize they will eventually move, but at least the copying gives a stable URL, thanks. :+1:

**Comment by richsalz** [comment]
Levitte might  come up with a different approach, but at least for now (until next release) I fixed it :)

**Comment by mattcaswell** [comment]
This was due to an error that occurred in the release today. The old links should work soon (they can take a couple of hours to propagate)

> The simplest thing for us to do, @levitte is to copy the new release to the old directory right away. I just did it by hand for all the release streams.

No need we have rewrite rules for all the releases so that they get a stable URL directly under source...they just got accidentally zapped during the release.

**Comment by richsalz** [comment]
The stable URL for all releases is https://www.openssl.org/source/old/...

What’s wrong with that?

**Comment by mattcaswell** [comment]
> The stable URL for all releases is https://www.openssl.org/source/old/...
> What’s wrong with that?

There's nothing wrong with that per se. I only closed this because the original problem is no longer an issue. We have existing stable links for all releases (directly under source) and they accidentally got removed during the release yesterday. I restored the old links so everything should be working as it was before. If we decide to provide *additional* stable links for all releases in source/old (even for the latest release) then I have no objection to that.
